### ISSUE TITLE
Cross-client test vectors for ContentKeys

### ISSUE THREAD

**Issue by victor-wei126** [ethereum/trin#197]
Create test vectors for encoding & decoding ContentKeys (like [here](https://github.com/ethereum/portal-network-specs/blob/master/portal-wire-test-vectors.md)) following #108.

**Comment by victor-wei126** [comment]
@njgheorghita Looking at the test vectors that were implemented for portal wire protocol messages, I see that the only difference between my current unit tests are that the vectors explicitly test encoding, expecting that a given input matches to a specific hex string. I don't do this currently, I only encode, then immediately decode, and check to see if the decoded key matches the original. So is the task at hand here then to 1) create a test vector spec, specifying inputs/outputs for each encoding/decoding of each content key, and then 2) tweaking my unit tests to explicitly test encoding?

I was under the impression that I had to create some completely new tests, but it seems "test vector" and "unit test" are somewhat synonymous?

Also, on the topic of explicitly testing encoding, is the reason that is necessary is because the encoding may be incorrect, but somehow the decoding is, and so we miss that error if we don't check for it? Sorry for many questions, just trying to understand the rationale behind what I'm doing
